dhis-2/dhis-test-integration/src/test/java/org/hisp/dhis/tracker/deduplication/PotentialDuplicateRemoveTrackedEntityTest.java

```diff
@@ -199,28 +200,23 @@ void shouldDeleteEnrollments() throws ForbiddenException, NotFoundException, Bad
     manager.update(duplicate);
     manager.update(control1);
     manager.update(control2);
-    assertNotNull(trackedEntityService.getTrackedEntity(UID.of(original)));
-    assertNotNull(trackedEntityService.getTrackedEntity(UID.of(duplicate)));
-    assertNotNull(trackedEntityService.getTrackedEntity(UID.of(control1)));
-    assertNotNull(trackedEntityService.getTrackedEntity(UID.of(control2)));
+    assertNotNull(trackedEntityService.getNewTrackedEntity(UID.of(original)));
+    assertNotNull(trackedEntityService.getNewTrackedEntity(UID.of(duplicate)));
+    assertNotNull(trackedEntityService.getNewTrackedEntity(UID.of(control1)));
+    assertNotNull(trackedEntityService.getNewTrackedEntity(UID.of(control2)));
```

Comment:
getNewTrackedEntity does not return null right? it throws an exception

we should also annotate the method with `@Nonnull` to indicate that

Reply:
It is already annotated with @Nonnull.
We could change the assertion to `assertDoesNotThrow` but I will postpone this when I create the `findTrackedEntity` method, then the assertion will be cleaner